core/coreobjects/include/coreobjects/property_object_impl.h

```diff
@@ -1745,11 +1754,61 @@ ErrCode GenericPropertyObjectImpl<PropObjInterface, Interfaces...>::beginUpdate(
 
     updateCount++;
 
-    return OPENDAQ_SUCCESS;
+    return daqTry([this] { callBeginUpdateOnChildren(); });
+}
+
+template <typename PropObjInterface, typename... Interfaces>
+void GenericPropertyObjectImpl<PropObjInterface, Interfaces...>::callBeginUpdateOnChildren()
+{
+    for (const auto& item : propValues)
+    {
+        const auto value = item.second;
+        if (value.assigned())
+        {
+            const auto propObj = value.template asPtrOrNull<IPropertyObject>(true);
+            if (propObj.assigned())
+            {
+                propObj.beginUpdate();
+            }
+        }
+    }
+}
+
+template <typename PropObjInterface, typename... Interfaces>
+void GenericPropertyObjectImpl<PropObjInterface, Interfaces...>::callEndUpdateOnChildren()
+{
+    for (const auto& item : propValues)
+    {
+        const auto value = item.second;
+        if (value.assigned())
+        {
+            const auto propObj = value.template asPtrOrNull<IPropertyObject>(true);
+            if (propObj.assigned())
+            {
+                propObj.endUpdate();
+            }
+        }
+    }
+}
```

Comment:
This behaviour is right now hidden from users - there was no API or documentation change that came with this addition. We should maybe consider implementing this as an optional parameter - eg. `beginUpdate(bool recursive = true)`. At least the javadoc documentation must explain the new behaviour.

Reply:
As agreed, a change to the Doxygen documentation is sufficient.